NH3 thermo and kinetic data additions

alongd opened the pull request:
Added the `NH3` therno library based on CCSD(T)-F12/cc-pVTZ-F12//B2PLYPD3/aug-cc-pVTZ computations (enthalpies taken from ATcT where available)

Made significant modifications and additions to the kinetic `promaryNitrogenLibrary` based on updated literature and based on the submitted "NH3-1" paper

Also added several forbidden species and made two minor modifications to the `primaryNitrogenLibrary` kinetic library

We should still add PDep reactions calculated by us on the N2H3 surface, and later we expect to have updated reactions of the N2H4 surface.

Points for discussion:
- There are many non-phisical species in the H/N/O space (with legitimate Lewis structure and Molecule representation, but which do not exist, see [here](https://doi.org/10.1002/cphc.202200373)). Should we keep adding them to the forbidden structures file (they were originally suggested by RMG and many made it into the core in early model generation iterations), or should we consider another approach?
- The Dean&Bozzelli library has an unreasonable rate for `NH2 + OH <=> NH(S} + H2O`. One updated source that studies all `NH2/NH3 + OH` reaction at all multiplicities is [this](https://doi.org/10.1016/j.chemphys.2018.03.022), though it is very confusing, and some of the rates are still very high. This is currently unresolved.

jonwzheng commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+    label="HNO + HNO <=> N2O + H2O",
+    kinetics=Arrhenius(A=(8.4e+8, 'cm^3/(mol*s)'), n=0.0, Ea=(13000, 'kJ/mol'),
+                       T0=(1, 'K'), Tmin=(450, 'K'), Tmax=(520, 'K')),
+    shortDesc=u"""[Tsang1991]""",
```
seems to be reaction # 11,11 in Table 2
They report the same A-factor, but they report Ea/R = 1561 (or 3101 cal/mol) instead of 13000, could you please  check this?

alongd replied:
Correct, thanks!

jonwzheng reviewed (changes requested):
Hi Alon, thank you and your team for this big update of the thermo database. It can be difficult and tedious work to search carefully through the literature - so once again, thank you for your effort!

Due to the large size of this PR, I will split my review into multiple stages. This will help me get feedback to you as soon as possible rather than all at once.

I double-checked the new data added in `primaryNitrogenLibrary`, `primaryNS`, `primaryH2O2` and `HydrazinePDep`. I started looking at `forbiddenStructures` but will continune in a future review. I haven't yet looked at `NH3` library. Here are my overall comments:

* We have some concerns about forbidding these structures; a species may exist even if it cannot be found at a certain QM LOT. If we forbid a structure that ends up being an important intermediate, this could dramatically - and silently - impair model performance. On Bill's suggestion, I've started commenting on those ones that may need additional discussion, as several pathways in principle seem reasonable for generating those species (and I will examine the rest of the reactions in a future review). Could you please comment about your thoughts on this issue?

* Many reactions had their kinetic data replaced with newer data, could you briefly comment on this - were such old data in serious error?

* Some of the data seem to be transcribed in error, please check out the individual line comments.

caochuangmit commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -4775,47 +4824,27 @@
-    shortDesc = u"""[Glarborg2021]""",
+    kinetics = Arrhenius(A=(1.02e+12, 'cm^3/(mol*s)'), n=0.166, Ea=(-938, 'cal/mol'),
+                          T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2500, 'K')),
+    shortDesc = u"""[Klippenstein2022]""",
```
I check the original literature source and think Ea of the reaction `NH2 + HO2 <=> HNO + H2O` should be -938 cal/mol instead of -1864 cal/mol. Also, I believe the unit of R in Arrhenius equations shouldbe J mol−1 K−1 rather than cal mol−1 K−1 as stated in Klippenstein's work, after a test for another reaction (NH2+HO2=NH3+O2), where I use Klippenstein's fitted kinetic parameters to compare with the curve in figure 4 in their work.

caochuangmit commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -4843,11 +4868,17 @@
-    shortDesc = u"""[Sarathy2022]""",
+    kinetics = Arrhenius(A=(2.19e+09, 'cm^3/(mol*s)'), n=0.791, Ea=(-2838, 'cal/mol'),
+                         T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(2500, 'K')),
+    shortDesc = u"""[Klippenstein2022]""",
```
Please check Ea. I think for NH2+HO2 reactions calculated in Klippenstein2022, The Ea should be directly taken from the equations given in the main text and then converted to nagative values, without considering any unit conversions.

alongd replied:
I agree, fixed, please see if correct now?

ChgCao replied:
I think A=2.19e+09 and n=0.791 are correct, but Ea should be -1428 as expressed in R1c instead of -2838. Please double-check it.

alongd commented on input/forbiddenStructures.py:
```diff
@@ -64,6 +64,23 @@
+4 H u0 p0 c0 {1,S}
+""",
+    shortDesc = u"""""",
+    longDesc =
```
It was determined in a previous work to be non-physical along with NNOOH
I added NNOOH now as well, and added a short justification for both of these molecules

alongd replied:
For `k2` in the table (LHS), B values are negative, the B values on the right are for `k5`

alongd commented on input/forbiddenStructures.py:
```diff
@@ -524,3 +540,3487 @@
+    shortDesc = u"""""",
+    longDesc =
+u"""
+A CBS-QB3 job was unable to optimize this structure, got NO + H2NO
```
We think that many -N-O- chains will be unstable (though not all).
There are many forbidden molecules here that were excluded based on one calculation, and more work should be done to verify. I'm OK with removing species that were ruled out based on just one computation. When generating the NH3 model, please check whether any of them are in the core or are in a PDep network that yielded a core reaction - if so an additional study should be done for them.

alongd commented on input/forbiddenStructures.py:
```diff
@@ -524,3 +540,3487 @@
+    shortDesc = u"""""",
+    longDesc =
+u"""
+An wb97xd/def2svp conformer search was unable to optimize any of the conformer, they break down into O2 + H2NN
```
This was probably copied over, so probably the missing H in the comment is my typo. I had many of these and tried to give a lead to future studies on these species

alongd commented on input/forbiddenStructures.py:
```diff
@@ -524,3 +540,3487 @@
+    shortDesc = u"""""",
+    longDesc =
+u"""
+A CBS-QB3 job was unable to optimize this structure, got HO2 + H2NO
```
Looks like my comments were inaccurate. I had many cases of these species and I may have written the comments in a rush. I'm OK with not merging these forbidden molecules, it's also not an elegant method to exclude them like that, perhaps a deeper study can come up with some rules of thumb and an on-the-fly QM analysis can validate whether they exist or not.

alongd commented on input/forbiddenStructures.py:
```diff
@@ -524,3 +540,3487 @@
+    shortDesc = u"""""",
+    longDesc =
+u"""
+A CBS-QB3 job was unable to optimize this structure, got H2O + HNN
```
The comment wasn't meant for what could form this allegedly non-existing, it is meant to describe what the DFT opt gave - the species broke into these two fragments.

alongd commented on input/kinetics/libraries/primaryH2O2/reactions.py:
```diff
@@ -5,487 +5,512 @@
+    shortDesc=u"""[Konnov2015]""",
+    longDesc=u"""
+    Table 2, Reaction X15 
+    No data was given, the rate is very low.
```
There was a change to the counting index above. number 41 is now number 40
The rate remained the same.

I renumbered now again this section to avoid skipping the running indices of 41, 42

Sorry that I didn't separate content changes from formatting and renumbering changes, I can try to separate it into two commits if it helps

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -1361,31 +1386,35 @@
-Rate taken from:
+CCSD(T)/CBS(TQ5)//CCSD(T)/aug-cc-pVTZ+d
+
+Also available from D&B, originally taken from:
```
Sure

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+    shortDesc=u"""[Huynh2019]""",
+    longDesc=
+    u"""
+    Table S4 gives the raw data
```
Correct. I made the comment clearer now

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+entry(
+    index=424,
+    label="NH2 + OH <=> NH2O + H",
+    kinetics=Arrhenius(A=(6.4e+07, 'cm^3/(mol*s)'), n=0, Ea=(77.1, 'kJ/mol'), T0=(1, 'K')),
```
Good catch, thanks!!

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+            Arrhenius(A=(2.14e-12, 'cm^3/(mol*s)'), n=-6.56, Ea=(3279, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(7.83e-08, 'cm^3/(mol*s)'), n=-5.29, Ea=(469, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(5.70e-05, 'cm^3/(mol*s)'), n=-4.49, Ea=(-1157, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(1.31e-03, 'cm^3/(mol*s)'), n=-4.11, Ea=(-1938, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
```
Correct, `n` values should be positive here, I corrected.
Ea signs look good to me (look at the LHS for k3, not RHS for k_total)

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+            Arrhenius(A=(4.07e+04, 'cm^3/(mol*s)'), n=2.36, Ea=(-354, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(2.43e+05, 'cm^3/(mol*s)'), n=2.15, Ea=(-759, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(1.21e+06, 'cm^3/(mol*s)'), n=1.97, Ea=(-1166, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(1.95e+06, 'cm^3/(mol*s)'), n=1.92, Ea=(-1312, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
```
All looks good to me here

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+    shortDesc = u"""[Mei2019]""",
+    longDesc =
+u"""
+Reinterpreted rate coefficient based on the Miller and Bowman's work (https://doi.org/10.1016/0360-1285(89)90017-8)
```
I referred to this text of Mei2019:
```
In this work, we estimated the rate constant of R7 from the evaluated result of Miller and Bowman [52]
with the energy barrier adjusted according to the calculated result of Klippenstein et al. [54].
The comparison of rate constants of R7 from different sources is shown in [Fig. 1](c).
```

The rate coeff is in the SI:
```
N2H2+M=NNH+H+M                                  3.80E+13	1.2	70100 ! PW ZXY_20190214 estimated from 1979 Miller PECS and the PES calculated by SJK2009 JPCA
```

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+entry(
+    index=435,
+    label="NH3 + NH2 <=> N2H3 + H2",
+    kinetics=Arrhenius(A=(1.5e+13, 'cm^3/(mol*s)'), n=0.0, Ea=(8012, 'cal/mol'),
```
I agree, my mistake.
Also, in the body of the paper that say `A<1.3e+13` and in the abstract `A<1.5e+13`, I'll adopt the lower number here

alongd reviewed (commented):
Thanks for the helpful comments, and for actually checking the sources! Much appreciated.
Sorry for this huge PR.
Regarding the forbidden species, I'm open to remove most of them

alongd commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -2276,24 +2300,32 @@
+            Arrhenius(A=(2.04e+34, 'cm^3/(mol*s)'), n=-8.34, Ea=(-3309, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(1.92e+35, 'cm^3/(mol*s)'), n=-8.36, Ea=(-3474, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(2.85e+36, 'cm^3/(mol*s)'), n=-8.40, Ea=(-3821, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
+            Arrhenius(A=(5.22e+37, 'cm^3/(mol*s)'), n=-8.46, Ea=(-4416, 'cal/mol'), T0=(1, 'K'), Tmin=(300, 'K'), Tmax=(3000, 'K')),
```
For `k2` in the table (LHS), B values are negative, the B values on the right are for `k5`

jonwzheng commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -7233,13 +7209,15 @@
     index=435,
     label="NH3 + NH2 <=> N2H3 + H2",
-    kinetics=Arrhenius(A=(1.5e+13, 'cm^3/(mol*s)'), n=0.0, Ea=(8012, 'cal/mol'),
+    kinetics=Arrhenius(A=(1.3e+13, 'cm^3/(mol*s)'), n=0.0, Ea=(58.8, 'kcal/mol'),
```
I think this should be 5.88 rather than 58.8 kcal/mol, could you please double check?

jonwzheng replied:
My apologies, you're right!

jonwzheng commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5565,1686 @@
+    label="NH2O + NO2 <=> HNO + HONO",
+    kinetics=Arrhenius(A=(7.95, 'cm^3/(mol*s)'), n=2.95, Ea=(-3293, 'cal/mol'),
+                       T0=(1, 'K'), Tmin=(400, 'K'), Tmax=(2000, 'K')),
+    shortDesc=u"""[Cavallotti2023]""",
```
data from this source look good to me!

jonwzheng commented:
Thanks for checking the review comments. There are still a few small loose ends in the new library data, but those are just a matter of checking the original sources. I checked the `NH3` library and it looks okay to me.

The main question right now is whether we should merge the forbidden structures. I see the merit in excluding those potentially problematic species, but I also worry that some of them are real intermediates that could play a key role. One option is to _not_ forbid those structures, and then in our subsequent modeling work, see if they form in any appreciable amounts, and perhaps do a deeper analysis at that point. Is that what you suggested?  

Either is okay with me, but since this would affect the modeling workflow for the ammonia project, we should discuss with the team.

alongd commented:
@jonwzheng, I addressed the last comment on the PNL. Please let me know whether there are additional comments or whether I missed anything. I truly appreciate the time and effort of double-checking the values in this library, I know this is a relatively large PR that requires some time and effort to review.

I removed most of the forbidden species, let only two that were studied seriously.

I squashed the modifications, I think the PR is ready to be merged, of course let me know if there are any additional comments or concerns.

jonwzheng commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5496,3 +5563,1666 @@
+)
+
+entry(
+    index=419,
```
My last comment is that the indexing changes since 417 and 418 were removed. I don't know whether this will make a practical difference, but best to make the indexing consistent anyway.

jonwzheng reviewed (approved):
Just one comment about the index changing. 
Otherwise, looks good to me!

ChgCao commented on input/kinetics/libraries/primaryNitrogenLibrary/reactions.py:
```diff
@@ -5402,20 +5445,44 @@
-    shortDesc = u"""[Dievart2020]""",
+    index=432,
+    label="N2H3 + NH2 <=> H2NN(S) + NH3",
+    duplicate=True,
```
There are two reactions (this reaction and the last reaction) with index=432. Please move this reaction to the end and change the index number.

alongd commented:
@ChgCao, I updated the indices and squashed, if the tests pass I think we should be good to go

jonwzheng commented:
LGTM! but given the large scope @ChgCao could you also take a final look? If you approve, I'll go ahead and merge it in.
